### ISSUE TITLE
git push/pull: Access denied

### ISSUE THREAD

**Issue by miko** [sameersbn/docker-gitlab#26]
Hello,
I run sameersbn/docker-gitlab container on top of shipyard as a shipyard application (so it has its own domain, and hipache acts as a proxy). Everything works OK except git push/pull, which gives this error for me:

```
$ git push -u origin master
Access denied.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

That happens for both repository and wiki pages. I can succesfully log in as root using the same ssh connection, so ssh tunnel is OK (and so ssh keys are working OK). I am starting container with:

```
rm -rf /opt/gitlab
mkdir -p /opt/gitlab/data
mkdir -p /opt/gitlab/mysql
docker run -d \
  -p 80 \
  -p MYIP:22:22 \
  -v /opt/gitlab/data:/home/git/data \
  -v /opt/gitlab/mysql:/var/lib/mysql \
  -e "SMTP_USER=gitlab" -e "SMTP_PASS=MYPASS" \
  -e "SMTP_HOST=MYSMTP" -e "SMTP_PORT=25" \
  -e "GITLAB_EMAIL=gitlab@MYDOMAIN" \
  -e "GITLAB_SUPPORT=gitlab@MYDOMAIN" \
  -e "GITLAB_HOST=gitlab.MYDOMAIN" \
  -name gitlab \
  sameersbn/gitlab
```

(where MY\* are substituted for privacy).
I tried this many times from scratch (empty external data directory, empty internal mysql database with external data directory) and always get the same result. I even tried setting the full path with:

```
git remote add origin git@gitlab.MYDOMAIN:/home/git/data/repositories/MYUSER/test.git
```

I am able to import a repo from github, but then cannot clone out of this local repo (getting "Access denied." again). Can you help me debug the problem?

**Comment by sajeer850** [comment]
@sameersbn  please help me i want to push the folder its possible on gitlab 
please help  me
i want to upload 2 folder to push

**Comment by tyler43636** [comment]
I came across this thread after moving my data and containers to a new server. It ended up being a permissions problem. 

I fixed it by running: (requires docker 1.3 or newer)
`
docker exec -it (container name) bash
`
`
chown -R git:git /home/git/data
`
